### ISSUE TITLE
[Accessibility]: Code Insights - Add insight to dashboard

### ISSUE THREAD

**Issue by unclejustin** [sourcegraph/sourcegraph#41764]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/33495

### Problem description

![PixelSnap 2022-09-19 at 15 28 20](https://user-images.githubusercontent.com/1855233/191110528-101444ac-c5bb-4985-80a9-a5b709c0641b.png)

When this modal opens, it announces as having a modal with "3" items. 

This is misleading.

### Expected behavior

While announcing 3 items is technically correct, that is not what this modal is trying to convey. It should announce that there is a list of insights in this modal. Without this announcement, the user would not know that there are selections they are expected to make.

Side note: Would we be better served using a multi-selectable combo box?

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by felixfbecker** [comment]
The behavior of announcing the number of children the dialog has is default VoiceOver behavior we cannot change. I think this is fine.

However, the dialog has other problems. The list of dashboards is not marked up as a list, and to get to "Cancel"/"Save", you need to skip through potentially hundreds of dashboards.

Justin is potentially right here that using an ARIA widget may be better here, but at least it should be marked up as a list.
